utils/utils.c

```diff
@@ -178,12 +178,45 @@ int remove_directory(char *dirname)
 	return ret;
 }
 
+#define PATH_SEPARATOR '/'
+
+int check_bin_under_path(char *bin_path, char *path)
+{
+        int bin_leng = strlen(bin_path);
+        int path_leng = strlen(path);
+        int index = 0;
+
+        if (path_leng == 1 && path[0] == PATH_SEPARATOR)
+		return -2;
+
+        if (bin_leng < path_leng) return -1;
+
+        while (index < path_leng) {
+                if (bin_path[index] != path[index])
+                        return -1;
+                index++;
+        }
+
+        if (bin_path[index] == PATH_SEPARATOR || bin_path[index] == '\00')
+                return 0;
+
+	return -1;
+}
```

Comment:
I'm not sure checking the path of uftrace is meaningful here.  It'll usually be in a read-only directory so I don't think we need to check it.  But I think we should check the given (data) directory is an ancestor of the **current** directory.

Reply:
@ParkHanbum We should consider its ancestor.  It will make a problem if you record a target program with uftrace as follows:
```
$ pwd
/home/honggyu/work/myprog
$ gcc -pg hello.c
```
If its ancestor directory is given with `-d` option, it looks the record is done without a problem.
```
$ uftrace record -d /home/honggyu/work a.out
Hello, World!
$ pwd
/home/honggyu/work/myprog
```
But if you go back to the parent directory, then you will see the directory name is change from `work` to `work.old`.
```
$ cd ..
$ pwd
/home/honggyu/work.old
```
We're thinking that it's the problematic situation, so please consider this as well.

Reply:
And previously existing `work.old` directory contains full of uftrace generated files, which looks very confusing to users.

Reply:
Now I think that we need to check it's actually a uftrace data directory first.